interactive_engine/executor/store/src/api/property.rs

```diff
@@ -44,6 +44,51 @@ pub enum Property {
     Unknown,
 }
 
+impl PartialOrd for Property {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        match (self, other) {
+            (Property::Bool(left), Property::Bool(right)) => left.partial_cmp(right),
+            (Property::Char(left), Property::Char(right)) => left.partial_cmp(right),
+            // cmp between integer and float
+            // if both are integers, cast to long
+            // else cast to double
+            (Property::Short(_), _)
+            | (Property::Int(_), _)
+            | (Property::Long(_), _)
+            | (_, Property::Short(_))
+            | (_, Property::Int(_))
+            | (_, Property::Long(_)) => {
+                if self.is_float_type() || other.is_float_type() {
+                    let left = self.get_double().expect("Convert error");
```

Comment:
It's better not to use `expect()` here, which will panic once Property is neither integers nor floats. Same as the follows.

Reply:
introduce a macro to handle it